### PR TITLE
Add test coverage for sync plan info functionality (BZ1390545)

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/cli/test_syncplan.py
.............EE.....
======================================================================
ERROR: Create a sync plan with sync date in a future and sync one RH
----------------------------------------------------------------------
CLIReturnCodeError: CLIReturnCodeError ... Error: Archive failed signature check
======================================================================

ERROR: Create a sync plan with past datetime as a sync date, add a
----------------------------------------------------------------------
CLIReturnCodeError: CLIReturnCodeError ... Error: Archive failed signature check

----------------------------------------------------------------------
Ran 20 tests in 2732.134s

FAILED (errors=2)
```

Few failures are not related to changes provided

### CODE REVIEW

File: tests/foreman/cli/test_syncplan.py [hunk 1]
```diff
@@ -365,6 +363,33 @@ def test_positive_info_enabled_field_is_displayed(self):
         result = SyncPlan.info({'id': new_sync_plan['id']})
         self.assertIsNotNone(result.get('enabled'))
 
+    @tier1
+    @upgrade
+    def test_positive_info_with_assigned_product(self):
+        """Verify that sync plan info command returns list of products that
+        assigned to that sync plan
```
Comment (abalakh) [inline tests/foreman/cli/test_syncplan.py]: nitpick: missing 'are' before `assigned`? anyway `list of products assigned` or `list of products which are assigned` sounds better imo

File: tests/foreman/cli/test_syncplan.py [hunk 2]
```diff
@@ -365,6 +363,33 @@ def test_positive_info_enabled_field_is_displayed(self):
...
+
+        :id: 7a7e5e40-7776-4d26-9173-73f00de6e8e9
+
+        :expectedresults: Expected product information is returned in info
+            command
+
+        :BZ: 1390545
+        """
+        sync_plan = self._make_sync_plan({
+            'enabled': 'false',
```
Comment (abalakh) [inline tests/foreman/cli/test_syncplan.py]: nitpick: you could use bool `False` instead of that string

File: tests/foreman/cli/test_syncplan.py [hunk 3]
```diff
@@ -365,6 +363,33 @@ def test_positive_info_enabled_field_is_displayed(self):
...
+            'organization-id': self.org['id'],
+            'sync-date': datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S"),
+        })
+        product = make_product({'organization-id': self.org['id']})
+        Product.set_sync_plan({
+            'id': product['id'],
+            'sync-plan-id': sync_plan['id'],
+        })
+        updated_plan = SyncPlan.info({'id': sync_plan['id']})
+        self.assertEqual(len(updated_plan['products']), 1)
```
Comment (abalakh) [inline tests/foreman/cli/test_syncplan.py]: Maybe it's worth checking whether a _list_, not just 1 field, is returned? You could assign 2 products and make sure both are present in output.

File: tests/foreman/cli/test_syncplan.py (hunk 2 above)
Comment (oshtaier) [reply tests/foreman/cli/test_syncplan.py]: getting exception on that, it seems that it is necessary change factory method signature or something. Anyway, skip at that moment

**Comment by oshtaier** [comment]
```
nosetests tests/foreman/cli/test_syncplan.py -m test_positive_info_with_assigned_product
.
----------------------------------------------------------------------
Ran 1 test in 47.996s

OK
```

**Pull request merged**